mbs_results/calculate_estimation_weights.py

```diff
@@ -30,60 +31,63 @@ def calculate_design_weight(
     """
-    population_counts = dataframe.groupby([period, strata]).size()
+    population_counts = population_frame.groupby([period, strata]).size()
 
-    sample = dataframe[dataframe[sampled] == 1]
+    sample = population_frame[population_frame[sampled] == 1]
     sample_counts = sample.groupby([period, strata]).size()
 
     design_weights = population_counts / sample_counts
 
     design_weights.name = "design_weight"
     design_weights = design_weights.reset_index()
 
-    dataframe = dataframe.merge(design_weights, how="left", on=[period, strata])
+    population_frame = population_frame.merge(
+        design_weights, how="left", on=[period, strata]
+    )
 
-    return dataframe
+    return population_frame
 
 
 def calculate_calibration_factor(
-    dataframe: pd.DataFrame,
+    population_frame: pd.DataFrame,
     period: str,
     group: str,
     sampled: str,
     auxiliary: str,
     design_weight: str,
+    **config,
 ) -> pd.DataFrame:
     """
-     Add column to dataframe to calculate calibration factor
-
-     Parameters
-     ----------
-     dataframe : pd.DataFrame
-         data to be weighted
-     period : str
-         name of column in dataframe containing period variable
+    Add column to dataframe to calculate calibration factor
+
+    Parameters
+    ----------
+    population_frame : pd.DataFrame
+        data to use to calculate weights
+    period : str
+        name of column in dataframe containing period variable
     group: str
-         name of column in dataframe containing group level data
-         for separate ratio, use strata variable
-         for combined ratio, use calibration group level variable
-     sampled : str
-         name of column in dataframe containing sample flag
-     auxiliary : str
-         name of column in dataframe containing auxiliary variable
-     design_weight: str
-         name of column in dataframe containing design weight
-
-     Returns
-     -------
-     pd.DataFrame
-         dataframe with new column `calibration_factor`
+        name of column in dataframe containing group level data
+        for separate ratio, use strata variable
+        for combined ratio, use calibration group level variable
+    sampled : str
+        name of column in dataframe containing sample flag
+    auxiliary : str
+        name of column in dataframe containing auxiliary variable
+    design_weight: str
+        name of column in dataframe containing design weight
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe with new column `calibration_factor`
     """
 
-    population_sums = dataframe.groupby([period, group])[auxiliary].sum()
+    population_sums = population_frame.groupby([period, group])[auxiliary].sum()
 
     # copy to avoid SettingWithCopy warning
     # (not required with later versions of pandas)
-    sample = dataframe.copy()[dataframe[sampled] == 1]
+    sample = population_frame.copy()[population_frame[sampled] == 1]
```

Comment:
Of course it is great, as it is and quite clear what is doing, i have been using .loc to set values to avoid warning

Reply:
Happy to make that change, this doesn't need to be a copy here for any reason other than the pandas ambiguity warning

Reply:
I changed the `.copy()[]` to `.loc[]` and it gives the setting with copy warning so this one might be a bit more tricky (in fact I think I remember it being especially awkward)